homeassistant/components/fan/__init__.py

```diff
@@ -180,45 +193,57 @@ def handle_fan_service(service: str) -> None:
     return True
 
 
-class FanEntity(Entity):
+class FanEntity(ToggleEntity):
     """Representation of a fan."""
 
     # pylint: disable=no-self-use, abstract-method
 
-    def set_speed(self: Entity, speed: str) -> None:
+    @property
+    def state(self: ToggleEntity) -> str:
+        """Get the state of the fan."""
+        return self.__getattribute__(ATTR_SPEED)
+
+    def set_speed(self: ToggleEntity, speed: str) -> None:
         """Set the speed of the fan."""
         pass
 
-    def turn_on(self: Entity, speed: str=None) -> None:
+    def turn_on(self: ToggleEntity, speed: str=None, **kwargs) -> None:
         """Turn on the fan."""
-        pass
+        raise NotImplementedError()
 
-    def turn_off(self: Entity) -> None:
+    def turn_off(self: ToggleEntity, **kwargs) -> None:
         """Turn off the fan."""
-        pass
+        raise NotImplementedError()
 
-    def oscillate(self: Entity) -> None:
+    def oscillate(self: ToggleEntity, oscillating: bool) -> None:
         """Oscillate the fan."""
         pass
 
+    def is_on(self):
+        """Return true if the entity is on."""
+        return self.state not in [STATE_OFF, STATE_UNKNOWN]
```

Comment:
Make speed attribute a property and add a custom UI for fan that shows the speed besides on/off.

Do you want to have the speed as state to be able to have a plot of the speed, like a sensor?